src/testing/deploy/upgrade_test.go

```diff
@@ -121,47 +155,74 @@ func TestUpgradeTrigger(t *testing.T) {
 				false,
 			))
 			for i := 0; i < 11; i++ {
-				require.NoError(t, c.PutFile(dataCommit, "/hello", strings.NewReader("hello world")))
+				require.NoError(t, c.PutFile(dataCommit, "/hello", strings.NewReader(fmt.Sprintf("hello world %v", i))))
 			}
-			latestDataCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "master", "")
 			require.NoError(t, err)
+			expectedCommitCount := getExpectedCommitCountFromVersion(from)
+			var buf bytes.Buffer
 			require.NoErrorWithinTRetry(t, 2*time.Minute, func() error {
-				ci, err := c.InspectCommit(pfs.DefaultProjectName, "TestTrigger2", "master", "")
+				commits, err := c.ListCommit(client.NewRepo(pfs.DefaultProjectName, pipeline1), nil, nil, 0)
 				require.NoError(t, err)
-				aliasCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "master", ci.Commit.Id)
 				if err != nil {
 					return err
 				}
-				if aliasCI.Commit.Id != latestDataCI.Commit.Id {
-					return errors.New("not ready")
+				t.Logf("comparing commit trigger1 sizes %d/%d", len(commits), expectedCommitCount.preTrigger1)
+				for i, commit := range commits {
+					c.GetFile(commit.Commit, "/hello", &buf)
+					t.Logf("	%d: %s", i, buf.String())
+					buf.Reset()
+				}
+				if got, want := len(commits), expectedCommitCount.preTrigger1; got < want {
+					return errors.Errorf("trigger1 not ready; got %v commits, want %v commits", got, want)
 				}
 				return nil
 			})
-		},
-		func(t *testing.T, ctx context.Context, c *client.APIClient, _ string) { /* postUpgrade */
-			for i := 0; i < 10; i++ {
-				require.NoError(t, c.PutFile(dataCommit, "/hello", strings.NewReader("hello world")))
-			}
-			latestDataCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "master", "")
-			require.NoError(t, err)
-			require.NoErrorWithinTRetryConstant(t, 5*time.Minute, func() error {
-				ci, err := c.InspectCommit(pfs.DefaultProjectName, "TestTrigger2", "master", "")
+			require.NoErrorWithinTRetry(t, 2*time.Minute, func() error {
+				commits, err := c.ListCommit(client.NewRepo(pfs.DefaultProjectName, pipeline2), nil, nil, 0)
 				require.NoError(t, err)
-				aliasCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "", ci.Commit.Id)
 				if err != nil {
 					return err
 				}
-				if aliasCI.Commit.Id != latestDataCI.Commit.Id {
-					return errors.Errorf("not ready alias commit: %v latest data commit: %v", aliasCI.Commit.Id, latestDataCI.Commit.Id)
+				t.Logf("comparing commit trigger2 sizes %d/%d", len(commits), expectedCommitCount.preTrigger2)
+				for i, commit := range commits {
+					c.GetFile(commit.Commit, "/hello", &buf)
+					t.Logf("	%d: %s", i, buf.String())
+					buf.Reset()
+				}
+				if got, want := len(commits), expectedCommitCount.preTrigger2; got < want {
```

Comment:
After everything is done with these retries we should verify that `got == want` as well for both of these pipelines as our test assertion. I kinda think we should time.sleep for 1 second before checking `==` but I'm on the fence.

Reply:
just gonna change it to `!=`. the commits usually populate in groups, so if our `got` gets large we're pretty likely to see it